website.Rmd

```diff
@@ -2,17 +2,396 @@
+#' I am a big fan of [thisfunction()] in my package. I
+#' also have something to say about [otherpkg::otherfunction()]
+#' in somebody else's package.
+```
+
+On pkgdown sites, those square-bracketed functions become hyperlinks to the relevant pages in your pkgdown site.
+This is automatic within your package.
+But links from *other* people's packages (and websites, etc.) require two things:
+
+-   The `URL` field of your `DESCRIPTION` file must include the URL of your pkgdown site (preferably followed by the URL of your GitHub repo):
+
+    ``` yaml
+    URL: https://dplyr.tidyverse.org, https://github.com/tidyverse/dplyr
+    ```
+
+-   Your `_pkgdown.yml` file must include the URL for your site:
+
+    ``` yaml
+    url: https://dplyr.tidyverse.org
+    ```
+
+devtools takes every chance it gets to do this sort of configuration for you.
+But if you elect to do things manually, this is something you might overlook.
+A general resource on auto-linking in pkgdown is `vignette("linking", package = "pkgdown")`.
+
+By default, the reference index is just an alphabetically-ordered list of functions.
+For packages with more than a handful of functions, it's often worthwhile to curate the index and organize the functions into groups.
+For example, dplyr uses this technique: <https://dplyr.tidyverse.org/reference/index.html>.
+You achieve this by providing a `reference` field in `_pkgdown.yml`.
+Here's a redacted excerpt from dplyr's `_pkgdown.yml` file that gives you a sense of what's involved:
+
+``` yaml
+reference:
+- title: Data frame verbs
+
+- subtitle: Rows
+  desc: >
+    Verbs that principally operate on rows.
+  contents:
+  - arrange
+  - distinct
+  ...
+
+- subtitle: Columns
+  desc: >
+    Verbs that principally operate on columns.
+  contents:
+  - glimpse
+  - mutate
+  ...
+
+- title: Vector functions
+  desc: >
+    Unlike other dplyr functions, these functions work on individual vectors,
+    not data frames.
+  contents:
+  - between
+  - case_match
+  ...
+
+- title: Built in datasets
+  contents:
+  - band_members
+  - starwars
+  - storms
+  ...
+
+- title: Superseded
+  desc: >
+    Superseded functions have been replaced by new approaches that we believe
+    to be superior, but we don't want to force you to change until you're
+    ready, so the existing functions will stay around for several years.
+  contents:
+  - sample_frac
+  - top_n
+  ...
+```
+
+To learn more, see `?pkgdown::build_reference`.
+
+## Vignettes and articles
+
+@sec-vignettes deals with vignettes, which are long-form guides for a package.
+They afford various opportunities beyond what's possible in function documentation.
+For example, it's much easier to work with code that produces an error, it's less cumbersome to work with suggested packages, and there are ways to shield the reader from setup and teardown that does not apply to authentic usage.
+
+A package's vignettes appear, in rendered form, in its website, in the "Articles" dropdown menu.
```

Comment:
Maybe work in something like: A vignette is technical term that we don't expect all R users to know; that's why pkgdown uses the more general articles here.

Reply:
I've added this point.